### PR TITLE
Session being updated incorrectly on Gateway mode

### PR DESCRIPTION
**Pull request by ealsur**
## Description

When dealing with Exception-less responses (Conflict, NotFound, Precondition failed, Throttling), the GatewayStoreModel was updating the session token when it shouldn't be.

This PR adds the fix plus tests for both Exceptions and Exception-less scenarios.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#705](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/705) User agent suffix gets truncated
 - [#753](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/753) Reason was not being propagated for Conflict exceptions
 - [#756](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/756) Change Feed Processor with WithStartTime would execute the delegate the first time with no items.
+- [#769](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/769) Gateway mode was updating Session Token incorrectly for some response types.
```
Comment (kirankumarkolli) [inline changelog.md]: SessionConsistency + Gateway mode session-token bug fix: Under few rare non-success cases session token might be in-correct

**Comment by kirankumarkolli** [comment]
For direct mode direct package upgrade is needed right?

**Comment by ealsur** [comment]
@kirankumarkolli Yes, Direct package upgrade is needed

File: changelog.md (hunk 1 above)
Comment (ealsur) [reply changelog.md]: Updated

**Pull request merged**